server.js

```diff
@@ -20,14 +27,112 @@ app.use(express.urlencoded({ extended: false }));
 // JSON
 app.use(express.json());
 
-app.use(express.static(path.join(__dirname, 'public')));
+// serve our images
+app.use('/images', express.static(path.join(__dirname, 'public/images')));
+
+function getMimeType (v, dft) {
+  try {
+    let ret;
+    if (mime.lookup) {
+      ret = mime.lookup(v) || dft || 'text/html';
+    } else {
+      ret = mime.getType(v) || dft || 'text/html';
+    }
+    return ret;
+  } catch (ex) {
+    return dft || 'text/html';
+  }
+}
+
+
+function removeBase (uri, base, noLeadingSlash) {
+  let ret;
+  if (!base || !base.length) {
+    ret = uri;
+  } else if (uri.length < base.length) {
+    ret = uri;
+  } else if (uri.substring(0,base.length).toLowerCase() !== base) {
+    ret = uri;
+  } else {
+    ret = uri.substring(base.length, uri.length);
+  }
+  
+  if (noLeadingSlash && ret.length > 0 && ret.substring(0,1) === '/') {
+    ret = ret.substring(1,ret.length);
+  }
+  
+  return ret;
+}
 
 
 
+async function start () {
+  const IMG_BASE = '/images';
+  const SCRIPT_BASE = '/script';
+  const VIEW_BASE = '/view';
+  
+  
+  const IMG_URL_PATH = new RegExp(`${IMG_BASE}\..*`,'i');
+  const SCRIPT_URL_PATH = new RegExp(`${SCRIPT_BASE}\..*`,'i');
+  const VIEW_URL_PATH = new RegExp(`${VIEW_BASE}\..*`,'i');
+  
+  
+  app.get(SCRIPT_URL_PATH, async (req, res, next) => {
+    try {
+      let name = removeBase(req.path,SCRIPT_BASE,true);
+      
+      let opfile = path.join(VIEW_PATH, name.trim());
+      if (opfile.slice(-3).toLowerCase() !== '.js') opfile+= '.js';
+      
+      
+      res.type(getMimeType(opfile))
+      res.sendFile(path.join(appRoot,`/public/script/${name}`));
+    } catch (e) {
+      console.log(e);
+    }
+  });
+  
+  app.get(VIEW_URL_PATH, async (req, res, next) => {
+    try {
+      let name = removeBase(req.path,VIEW_BASE,true);
+      let opfile = path.join(VIEW_PATH, name.trim());
+      
+      if (opfile.slice(-5).toLowerCase() !== '.html') opfile+= '.html';
+      
+      
+      res.type(getMimeType(opfile))
+      res.sendFile(path.join(appRoot,`/public/view/${name}.html`), (err)=>{
```

Comment:
If I type sitename.com/view/register.html should that also work, or only view/register? I see the conditional to add it to opfile if it is missing, but then it is hardcoded when sending the response.